fdbserver/ClusterController.actor.cpp

```diff
@@ -2545,23 +2548,8 @@ ACTOR Future<Void> clusterControllerCore(Reference<IClusterConnectionRecord> con
 		}
 		when(state RegisterWorkerRequest req = waitNext(interf.registerWorker.getFuture())) {
 			++self.registerWorkerRequests;
-			state ClusterConnectionString ccs = coordinators.ccr->getConnectionString();
-
-			state std::unordered_set<NetworkAddress> coordinatorAddresses;
-			std::vector<Future<Void>> fs;
-			for (auto& hostname : ccs.hostnames) {
-				fs.push_back(map(hostname.resolve(), [&](Optional<NetworkAddress> const& addr) -> Void {
-					if (addr.present()) {
-						coordinatorAddresses.insert(addr.get());
-					}
-					return Void();
-				}));
-			}
-			wait(waitForAll(fs));
-
-			for (const auto& coord : ccs.coordinators()) {
-				coordinatorAddresses.insert(coord);
-			}
+			ClusterConnectionString ccs = coordinators.ccr->getConnectionString();
+			state std::vector<NetworkAddress> coordinatorAddresses = wait(ccs.tryResolveHostnames());
```

Comment:
The `wait` here will block all other `when` cases, so it's undesirable. Should do something like
```
self.addActor.send(registerWorker(...));
```
where `registerWorker` is an actor that calls the `wait` internally.

Reply:
Thank you, done. Fixed in #6883 too.

Reply:
This kind of code could be more reliably detected if we implemented a warning in Actor compiler, i.e. emit a warning when there is a `wait` inside `when` https://github.com/apple/foundationdb/issues/1502

Reply:
@jzhou77 ^^